core/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java

```diff
@@ -34,17 +34,32 @@ public interface PhysicsRollingStock {
 
     /** Get the effort the train can apply at a given speed, in newtons */
     static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCurve) {
-        double previousEffort = 0.0;
-        double previousSpeed = 0.0;
-        for (var dataPoint : tractiveEffortCurve) {
-            if (previousSpeed <= Math.abs(speed) && Math.abs(speed) < dataPoint.speed()) {
-                var coeff = (previousEffort - dataPoint.maxEffort()) / (previousSpeed - dataPoint.speed());
-                return previousEffort + coeff * (Math.abs(speed) - previousSpeed);
+        int index = 0;
+        int left = 0;
+        int right = tractiveEffortCurve.length - 1;
+        while (left <= right) {
+            int mid = (left + right) >>> 1;
+            if (Double.compare(tractiveEffortCurve[mid].speed, Math.abs(speed)) == 0) {
```

Comment:
This doesn't really solve the "float comparaison" problem, it just hides it from spotbugs. If you think an equality test actually makes sense here, the correct way to remove the warning is to add `@SuppressFBWarnings({"FE_FLOATING_POINT_EQUALITY"})` before the method.

There's probably a way to remove this `if` branch, but that's not really important.